test/addon/fake_process_server.rb

```diff
@@ -0,0 +1,29 @@
+# typed: true
+# frozen_string_literal: true
+
+require "json"
+require "ruby_lsp/addon/process_server"
+
+module RubyLsp
+  class Addon
+    class FakeProcessServer < ProcessServer
+      def generate_initialize_response
+        JSON.dump({ result: { initialized: true } })
+      end
+
+      def execute(request, params)
+        case request
+        when "echo"
+          { result: { echo_result: params[:message] } }
+        when "shutdown"
+          @running = false
+          { result: {} }
+        else
+          VOID
```

Comment:
We shouldn't need the `VOID` stuff any more now that https://github.com/Shopify/ruby-lsp-rails/pull/453 is merged.